lib/molinillo/resolution.rb

```diff
@@ -364,6 +364,16 @@ def fixup_swapped_children(vertex)
             debug(depth) { "Removing orphaned spec #{succ.name} after swapping #{name}" }
             activated.detach_vertex_named(succ.name)
             requirements.delete_if { |r| name_for(r) == succ.name }
+            # TODO:
+            # 1. I'm not sure if it's correct to be deleting these
+            # requirements. Does not supporting circular deps ensure that the
+            # requirement must have come from `succ`? What about diamond-shaped
+            # dependency graphs?
+            # 2. If it's okay to delete the requirements, this should be
+            # recursive.
+            succ.successors.each do |succ_dep|
```

Comment:
Definitely successors. In my repro case, I have: `build-essential(new) -> seven_zip -> windows` and `build-essential(old) -> 7-zip -> windows`. It's when Molinillo detects that it needs to replace `build-essential(new)` with `build-essential(old)` that we hit this code path; in this case, `windows` is the "zombie" requirement, it's a successor of `seven_zip`.

And yeah, per the comments above, this is a bad patch as currently written, I just wanted to get your opinion on the general approach so I didn't waste effort on doing the wrong thing. I'll re-write it.